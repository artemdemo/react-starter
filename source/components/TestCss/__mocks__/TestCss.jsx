import React from 'react';

const TestCss = () => (
    <div data-mock='TestCss' />
);

export default TestCss;
