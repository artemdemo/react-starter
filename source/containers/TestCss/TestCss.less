.test-css {
    display: inline-block;
    background-color: #E3F2FD;
    border: 1px solid #BBDEFB;
    border-bottom: 3px solid #BBDEFB;
    border-top-left-radius: 3px;
    border-top-right-radius: 3px;
    padding: 5px 5px;
}

.test-css__mini {
    background-color: #8BC34A;
    color: white;
    display: inline-block;
    padding: 0 5px;
    border-radius: 2px;

    &:hover {
        background-color: #689F38;
    }
}
