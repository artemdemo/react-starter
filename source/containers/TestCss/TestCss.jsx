import React from 'react';

import './TestCss.less';

const TestCss = () => (
    <div className='test-css'>
        This is custom component that helps test <span className='test-css__mini'>css minification</span>.
    </div>
);

export default TestCss;
