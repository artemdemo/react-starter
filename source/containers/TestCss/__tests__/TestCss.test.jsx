import React from 'react';
import renderer from 'react-test-renderer';
import TestCss from '../TestCss';

jest.mock('../../../components/Select/Select');

describe('<TestCss>', () => {
    it('Simple render', () => {
        const tree = renderer.create(
            <TestCss />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
