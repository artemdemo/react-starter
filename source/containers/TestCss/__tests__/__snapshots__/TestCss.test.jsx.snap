// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestCss> Simple render 1`] = `
<div
  className="row"
>
  <div
    className="col"
  >
    <div
      data-mock="Select"
    >
      {
  "value": {
    "value": "show-css",
    "name": "Show CSS message"
  },
  "list": [
    {
      "name": "Select message"
    },
    {
      "value": "show-css",
      "name": "Show CSS message"
    },
    {
      "value": "hide-css",
      "name": "Hide CSS message"
    }
  ]
}
    </div>
  </div>
  <div
    className="col"
  >
    <div
      className="test-css"
    >
      This is custom component that helps testÂ 
      <span
        className="test-css__mini"
      >
        css minification
      </span>
      .
    </div>
  </div>
</div>
`;
